File: Java/src/net/mirchs/dm8/graphics/buffers/IndexBuffer.java
package net.mirchs.dm8.graphics.buffers;

import static org.lwjgl.opengl.GL15.*;

import net.mirchs.dm8.utils.BufferUtils;

public class IndexBuffer {

	private int ID;
	private int count;

	public IndexBuffer(int[] data) {
		count = data.length;
		
		ID = glGenBuffers();
		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ID);
		glBufferData(GL_ELEMENT_ARRAY_BUFFER, BufferUtils.createIntBuffer(data), GL_STATIC_DRAW);
		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);

	}

	public void bind() {
		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ID);
	}

	public void unbind() {
		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
	}
	
	public int getCount() {
		return count;
	}
}

File: C++/DM8/src/graphics/buffers/IndexBuffer.cpp
#include "IndexBuffer.h"

namespace DM8 {
	namespace graphics {

		IndexBuffer::IndexBuffer(unsigned int* data, unsigned int count) : 
			m_Count(count)
		{
			glGenBuffers(1, &m_ID);
			glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_ID);
			glBufferData(GL_ELEMENT_ARRAY_BUFFER, count, data, GL_STATIC_DRAW);
			glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
		}

		void IndexBuffer::Bind() const
		{
			glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_ID);

		}

		void IndexBuffer::Unbind() const
		{
			glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
		}


	}
}